test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -928,7 +964,12 @@ void UcxConnection::stream_recv_callback(void *request, ucs_status_t status,
 
     if (!conn->is_established()) {
         assert(conn->_establish_cb == r->callback);
-        conn->established(status);
+
+        if (status == UCS_OK) {
+            conn->established(status);
+        } else {
+            conn->handle_connection_error(status);
+        }
```

Comment:
why needed?

Reply:
to remove from in-progress list and conns map in case of error

Reply:
why not call "established" with error? this is what happened before when connection establish failed

Reply:
also seems like handle_connection_error invokes establish_cb itself

Reply:
> also seems like handle_connection_error invokes establish_cb itself

yes, correct
but also need to remove from in-progress list